lib/puppet/provider/package/sensu_gem.rb

```diff
@@ -15,6 +15,9 @@
     if RUBY_PLATFORM =~ /cygwin|mswin|mingw|bccwin|winse|emx/
       "#{ENV['SYSTEMDRIVE']}\\opt\\sensu\\embedded\\bin\\gem.cmd"
     else
+      # grab the newest version available of the embedded ruby
+      embedded_rvm_version = Dir.glob(File.join('/opt/sensu/embedded/lib/ruby/gems/', '*')).max { |a,b| File.ctime(a) <=> File.ctime(b) }
+      ENV['GEM_PATH'] = "/opt/sensu/embedded/lib/ruby/gems/#{embedded_rvm_version}"
```

Comment:
Disclaimer I come from a chef world so I have very little idea on the internals of puppet. Would the ENV var be passed down to where it actually executes it? This function just gets the command not runs it.

Reply:
I also was thinking about, "where should I put this?" My gut feeling is that if we have to use the embedded ruby, it's also appropriate to setup the ENV. Conversely we would not want to change the environment if we weren't using the sensu_gem provider. This sets up the environment only for the context of puppet, this isn't a system-wide change, so all the RVM stuff will remain untouched for everyone else.

Reply:
Right but is that for the entire puppet process or just for this particular resource. I worry that it might interfere if say you are installing a system wide gem with rvm in puppet. Another question is chef also has it's own embedded ruby which I am not sure if puppet does, can you install gems in puppet into it's own embedded gems? If so this might also break that as well. Hopefully someone with more puppet experience than me can review and provide some better feedback.